apps/sqlparser/sqlparser.py

```diff
@@ -0,0 +1,178 @@
+import argparse
+import shutil
+import csv
+import sys
+
+from datetime import datetime
+from pathlib import Path
+from collections import Counter
+
+# constants, can be overriden using arguments
+
+## implied folder structure: modules/*/sql/dbname/*/*.sql
+glob = "**/sql/**/*.sql"
+
+repo_folder = Path("/Users/v0/Py/_hobby/wow/server")
+
+
+default_argparsed_args = ["--help"]
+# default_argparsed_args = "-y", ""  # , "--glob", "test"
+
+source_suffix = "modules"
+source_dir = lambda: repo_folder / source_suffix
+target_suffix = "data/sql/custom"
+target_dir = lambda: repo_folder / target_suffix
+report_suffix = "modules_sql_parsed.csv"
+report_path = lambda: repo_folder / report_suffix
+
+modules_to_ignore = ["mod-playerbots"]
+
+counters = dict(
+    written=0, skipped=0, total=0, modules=Counter()  # module: counter
+)
```

Comment:
why `Counter()` only for `modules`? I assume since the `pathlib.Path` need to be hashed?

it might be more clear to split this out into a few variables instead of a dict:

```suggestion
sql_files_written = 0
sql_files_skipped = 0
modules_counter = Counter()
```

I think `total` could just be `written` + `skipped`? Not sure, worth a poke.

I probably sound like a broken record but globals can/should be avoided if possible

Reply:
regarding total, you are not 100% right.
total is pulled from report, and is basically an global index for all migrations.
its value is in the fact that you have a global index showing all migrations order.
therefore, once you have 1 entry in report, it will not match with written and skipped, which describe current script run.


all counters keys have very close purpose.
is there any value in splitting them? 
it will surely reduce readability and bloat the function annotations and returns.